1/backend/src/index.js

```diff
@@ -1,19 +1,35 @@
+const neo4j = require('neo4j-driver');
 const { ApolloServer } = require('apollo-server');
 const jwt = require('jsonwebtoken');
 const fs = require('fs');
+const { getDriver } = require('./neo4j.js')
 
 var SECRET_KEY = fs.readFileSync('./src/key/secret.key', 'utf8');
 
 const typeDefs = require('./schema');
-const resolvers = require('./resolvers')
+const resolvers = require('./resolvers');
+
+const driver = getDriver()
 
 const server = new ApolloServer({
   typeDefs,
   resolvers,
-  context: ({ req}) => {
-    req_header = req.headers.authorization;
-    if (req_header){
-      return {token: jwt.verify(req_header, SECRET_KEY)}
+  context: async ({ req}) => {
+    let token = req.headers.authorization
+    if (token){
+      token = token.replace('Bearer ', '')
+      try {
+        token = {token: jwt.verify(token, SECRET_KEY)}
+      }
+      catch(err) {
+        token = null
+      }
+    } else {
+      token = null
+    }
+    return {
+      driver,
+      token
     }
```

Comment:
Try to avoid nested if-clauses:
```js
const { authorization = '' } = req.header
let token = null
try {
  token = jwt.verify(token, SECRET_KEY)
} catch {
  // do nothing
}
return { driver, token }
```